sympy/calculus/codomain.py

```diff
@@ -151,3 +154,98 @@ def codomain_interval(f, set_val, *sym):
     if isinstance(domain, FiniteSet):
         return FiniteSet(*[limit(func, symbol, i) if i in FiniteSet(-oo, oo)
                             else func.subs({symbol: i}) for i in domain])
+
+
+def not_empty_in(finset_intersection, *syms):
+    """ Finds the domain of the functions in `finite_set` in which the
+    `finite_set` is not-empty
+
+    Parameters
+    ==========
+
+    finset_intersection: The unevaluated intersection of FiniteSet containing
+                        real-valued functions with Union of Sets
+    syms: Tuple of symbols
+            Symbol for which domain is to be found
+
+    Raises
+    ======
+
+    NotImplementedError
+        The algorithms to find the non-emptiness of the given FiniteSet are
+        not yet implemented.
+    ValueError
+        The input is not valid.
+    RuntimeError
+        It is a bug, please report to the github issue tracker.
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, codomain, FiniteSet, Interval, sqrt, oo
+    >>> from sympy.calculus.codomain import not_empty_in
+    >>> x = Symbol('x')
+    >>> not_empty_in(FiniteSet(x/2).intersect(Interval(0, 1)), x)
+    [0, 2]
+    >>> not_empty_in(FiniteSet(x, x**2).intersect(Interval(1, 2)), x)
+    [-sqrt(2), -1] U [1, 2]
+    >>> not_empty_in(FiniteSet(x**2/(x + 2)).intersect(Interval(1, oo)), x)
+    (-2, -1] U [2, oo)
+    """
+
+    # TODO: handle piecewise defined functions
+    # TODO: handle transcendental functions
+    # TODO: handle multivariate functions
+    if len(syms) == 0:
+        raise ValueError("A Symbol or a tuple of symbols must be given \
+                            as the third parameter")
+    if isinstance(finset_intersection, FiniteSet):
+        finite_set = finset_intersection
+        sets = S.Reals
+    elif isinstance(finset_intersection, Union):
+        elm_in_sets = finset_intersection.args[0]
+        return Union(not_empty_in(finset_intersection.args[1], *syms), elm_in_sets)
+    else:
+        finite_set = finset_intersection.args[1]
+        sets = finset_intersection.args[0]
+    if not isinstance(finite_set, FiniteSet):
+        raise ValueError('A FiniteSet must be given, not %s: %s' % (type(finite_set), finite_set))
+    if len(syms) == 1:
+        symbol = syms[0]
```

Comment:
`symbol` is a module in SymPy, you shouldn't override it. `sym = syms[0]` should work.